addons/account/models/account_move.py

```diff
@@ -3498,11 +3498,22 @@ def _post(self, soft=True):
             if wrong_lines:
                 wrong_lines.write({'partner_id': invoice.commercial_partner_id.id})
 
+        # reconcile if state is in draft and move has reversal_entry_id set
+        draft_refunds = to_post.filtered(lambda refund: refund.reversed_entry_id and refund.state == 'draft')
+
         to_post.write({
             'state': 'posted',
             'posted_before': True,
         })
 
+        for move, reverse_move in zip(draft_refunds.reversed_entry_id, draft_refunds):
+            group = (move.line_ids + reverse_move.line_ids) \
+                .filtered(lambda l: not l.reconciled) \
+                .grouped(lambda l: (l.account_id, l.currency_id))
+            for (account, _currency), lines in group.items():
+                if account.reconcile or account.account_type in ('asset_cash', 'liability_credit_card'):
+                    lines.with_context(move_reverse_cancel=True).reconcile()
```

Comment:
```suggestion
        for move, reverse_move in zip(draft_refunds.reversed_entry_id, draft_refunds):
            group = (move.line_ids + reverse_move.line_ids) \
                .filtered(lambda l: not l.reconciled) \
                .grouped(lambda l: (l.account_id, l.currency_id))
            for (account, _currency), lines in group.items():
                if account.reconcile or account.account_type in ('asset_cash', 'liability_credit_card'):
                    lines.reconcile()
```
This code is very similar (if not identical) to what is done in `_reverse_moves()`.
So instead of duplicating it, you might as well extract that logic in a dedicated method that you can call in both instances.
And forcing `move_reverse_cancel` to ´True` should not be necessary but could be problematic.